tools/topology/topology2/include/components/igo_nr.conf

```diff
@@ -0,0 +1,91 @@
+#
+#
+# A generic IGO_NR component. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.igo_nr.attribute_name"
+#
+# Usage: this component can be used by declaring in the parent object. i.e.
+#
+# Object.Widget.igo_nr."N" {
+#		index			1
+#	}
+# }
+
+#
+# Where M is pipeline ID and N is a unique integer in the parent object.
+
+<include/controls/mixer.conf>
+
+Class.Widget."igo_nr" {
+	#
+	# Pipeline ID
+	#
+	DefineAttribute."index" {
+		type "integer"
+	}
+
+	#
+	# Unique instance for IGO_NR widget
+	#
+	DefineAttribute."instance" {
+		type "integer"
+	}
+
+	# Include common widget attributes definition
+	<include/components/widget-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+			"instance"
+		]
+		!mandatory [
+			"num_input_pins"
+			"num_output_pins"
+			"num_input_audio_formats"
+			"num_output_audio_formats"
+		]
+
+		!immutable [
+			"uuid"
+			"type"
+		]
+		!deprecated [
+			"preload_count"
+		]
+		unique	"instance"
+	}
+
+	#
+	# IGO_NR widget switch control
+	#
+	Object.Control {
+		# Switch controls
+                mixer."1" {
+                        Object.Base.channel.1 {
+                                name    "fl"
+                                shift	0
+                        }
+                        Object.Base.channel.2 {
+                                name    "fr"
+                        }
+                        Object.Base.ops.1 {
+                                name	"ctl"
+                                info	"volsw"
```

Comment:
add comment for what "volsw" is, for all abbreviation names that not so obvious to know the meanning.

Reply:
@ranj063 @jsarha Do you know where this info is used? This has been "volsw" in all switch controls I've seen, so I've just repeated it. The control name for user space is defined elsewhere. Does the user see somehow this info text?

Reply:
The legal values are "volsw", "volsw_sx" and "volsw_xr_sx", and the value describes the kcontrol type of the mixer. It somehow describes how the mixer values are interpreted. The relevant functions are implemented in sound/soc/soc-ops.c, but I can not immediately say what is the relevant difference, or find any piece of documentation describing it.

Reply:
Thanks @jsarha I'll check if I can understand what they are and add a comment. This has not been clear to me either, I've just copied working topology examples from others.

Reply:
Volsw is the single or double mixer control. The _sx version is for use where signed values can be both sides of zero. _xr_sx is some multi-register to single combine control. I'll add some comment but it's not going to be well descriptive. E.g. ALSA documentation is not describing these, only lists the operations names @jsarha mentioned above.